pkg/plugins/config/config.go

```diff
@@ -31,13 +33,15 @@ type Cfg struct {
 	LogDatasourceRequests bool
 
 	PluginsCDNURLTemplate string
+
+	Opentelemetry tracing.OpentelemetryCfg
```

Comment:
Nit. We should maybe make this a pkg/plugins specific struct to remove dependency from pkg/infra/tracing. Lets hear what @wbrowne has to say about it.

Reply:
Oh yes plz 🙏

Reply:
@xnyo have you considered making this change?

Reply:
I am not sure how to do it properly to remove the dependency completely:

NewOpentelemetryCfg requires parseSettingsOpentelemetry, which is part of pkg/tracing

See here:

https://github.com/grafana/grafana/blob/giuseppe/plugins-backend-tracing/pkg/infra/tracing/config.go#L21-L35

I am not sure how to keep everything in pkg/plugins, without either:
1. duplicating the [config parsing code](https://github.com/grafana/grafana/blob/giuseppe/plugins-backend-tracing/pkg/infra/tracing/opentelemetry_tracing.go#L82-L115) (which could be done since it's only otel specific, but I would avoid it)
2. having something in pkg/infra/tracing that returns a new pkg/plugins.OpentelemetryCfg, but that would just invert the dependency (pkg/infra/tracing would depend on pkg/plugins)
3. moving the config provider to a separate package, which will be required by both pkg/infra/tracing and pkg/plugins. This is probably the cleanest solution but would require changes in pkg/infra/tracing as well

WDYT?

Reply:
I think what we could do here is actually change and move things to where they will end up eventually:

`pkg/plugins/config`:

```go
package config

import (
	"github.com/grafana/grafana-azure-sdk-go/azsettings"

	"github.com/grafana/grafana/pkg/plugins/log"
	"github.com/grafana/grafana/pkg/setting"
)

type Cfg struct {
	log log.Logger

	DevMode bool

	PluginsPath string

	PluginSettings       setting.PluginSettings
	PluginsAllowUnsigned []string

	// AWS Plugin Auth
	AWSAllowedAuthProviders []string
	AWSAssumeRoleEnabled    bool

	// Azure Cloud settings
	Azure *azsettings.AzureSettings

	BuildVersion string // TODO Remove

	LogDatasourceRequests bool

	PluginsCDNURLTemplate string

	OTel OTel
}

func NewCfg(devMode bool, pluginsPath string, pluginSettings setting.PluginSettings, pluginsAllowUnsigned []string,
	awsAllowedAuthProviders []string, awsAssumeRoleEnabled bool, azure *azsettings.AzureSettings, grafanaVersion string,
	logDatasourceRequests bool, pluginsCDNURLTemplate string, oTel OTel) *Cfg {
	return &Cfg{
		log:                     log.New("plugin.cfg"),
		PluginsPath:             pluginsPath,
		BuildVersion:            grafanaVersion,
		DevMode:                 devMode,
		PluginSettings:          pluginSettings,
		PluginsAllowUnsigned:    pluginsAllowUnsigned,
		AWSAllowedAuthProviders: awsAllowedAuthProviders,
		AWSAssumeRoleEnabled:    awsAssumeRoleEnabled,
		Azure:                   azure,
		LogDatasourceRequests:   logDatasourceRequests,
		PluginsCDNURLTemplate:   pluginsCDNURLTemplate,
		OTel:                    oTel,
	}
}


// Maybe place in a `pkg/plugins/config/tracing.go`
type OTel struct {
	Address     string
	Propagation string
}
```

`pkg/services/pluginsintegration/config`
```go
package config

import (
	"fmt"
	"strings"

	"github.com/grafana/grafana/pkg/infra/tracing"
	pconfig "github.com/grafana/grafana/pkg/plugins/config"
	"github.com/grafana/grafana/pkg/setting"
)

func ProvideConfig(settingProvider setting.Provider, grafanaCfg *setting.Cfg) (*pconfig.Cfg, error) {
	aws := settingProvider.Section("aws")
	plugins := settingProvider.Section("plugins")

	allowedUnsigned := grafanaCfg.PluginsAllowUnsigned
	if len(plugins.KeyValue("allow_loading_unsigned_plugins").Value()) > 0 {
		allowedUnsigned = strings.Split(plugins.KeyValue("allow_loading_unsigned_plugins").Value(), ",")
	}

	allowedAuth := grafanaCfg.AWSAllowedAuthProviders
	if len(aws.KeyValue("allowed_auth_providers").Value()) > 0 {
		allowedUnsigned = strings.Split(settingProvider.KeyValue("plugins", "allow_loading_unsigned_plugins").Value(), ",")
	}

	otelCfg, err := tracing.NewOpentelemetryCfg(grafanaCfg)
	if err != nil {
		return nil, fmt.Errorf("new opentelemetry cfg: %w", err)
	}
	return pconfig.NewCfg(
		settingProvider.KeyValue("", "app_mode").MustBool(grafanaCfg.Env == setting.Dev),
		grafanaCfg.PluginsPath,
		extractPluginSettings(settingProvider),
		allowedUnsigned,
		allowedAuth,
		aws.KeyValue("assume_role_enabled").MustBool(grafanaCfg.AWSAssumeRoleEnabled),
		grafanaCfg.Azure,
		grafanaCfg.BuildVersion,
		grafanaCfg.PluginLogBackendRequests,
		grafanaCfg.PluginsCDNURLTemplate,
		pconfig.OTel{
			Address:     otelCfg.Address,
			Propagation: otelCfg.Propagation,
		},
	), nil
}

func extractPluginSettings(settingProvider setting.Provider) setting.PluginSettings {
	ps := setting.PluginSettings{}
	for sectionName, sectionCopy := range settingProvider.Current() {
		if !strings.HasPrefix(sectionName, "plugin.") {
			continue
		}
		// Calling Current() returns a redacted version of section. We need to replace the map values with the unredacted values.
		section := settingProvider.Section(sectionName)
		for k := range sectionCopy {
			sectionCopy[k] = section.KeyValue(k).MustString("")
		}
		pluginID := strings.Replace(sectionName, "plugin.", "", 1)
		ps[pluginID] = sectionCopy
	}

	return ps
}
```

It's annoying to make that change here probably so we could do it as a follow up so that when the runner updates, it doesn't have to depend on that Grafana package

Reply:
Ah yes, makes sense. I can move `OpentelemetryCfg` to pkg/plugins for now (so it's clear it's something plugins-specific), then I can do that in a follow-up PR to move the initialization pluginsintegration (ProvideConfig) if that's okay?

Reply:
Perfecto 👍

Reply:
Good idea :+1:

Reply:
Done :)